sunpy/util/tests/test_metadata.py

```diff
@@ -0,0 +1,225 @@
+import pytest
+
+from sunpy.util.metadata import MetaDict
+
+
+def check_contents(metadict_inst, expected):
+    assert len(metadict_inst) == len(expected)
+
+    for key, val in expected:
+        assert metadict_inst[key.upper()] == val
+
+
+def check_insertion_order(metadict_inst, expected):
+
+    def normalise_keys(lst_pairs):
+        return [key.lower() for key, _ in lst_pairs]
+
+    assert normalise_keys(metadict_inst.items()) == \
+        normalise_keys(expected)
+
+
+def check(metadict_inst, expected):
+    """ Check a MetaDict instance
+
+    actual against expected contents and ensure insertion order is
+    preserved.
+
+    :param metadict_inst: a MetaDict
+    :param expected: iterable of key/value pairs
+    """
+    check_contents(metadict_inst, expected)
+    check_insertion_order(metadict_inst, expected)
+
+
+def pairs_to_dict(lst_of_pairs):
+    """ convert a list/tuple of lists/tuples to a dictionary
+
+    E.g.  [['a', 1], ['b', 2]] -> {'a': 1, 'b': 2}
+    """
+    return {kv_pair[0]: kv_pair[1] for kv_pair in lst_of_pairs}
+
+
+@pytest.fixture
+def std_test_data():
+    return [['labrador', 'americas'],
+            ['Norwegian', 'Europe'],
+            ['BALTIC', 'europe'],
+            ['LaPteV', 'arctic']]
+
+
+@pytest.fixture
+def differet_keys_data():
+    """
+    Contains distinct keys from 'std_test_data'
+    """
+    return [['Aegean', 'greece'],
+            ['CAtaLaN', 'mediterranean'],
+            ['IONIAN', 'italy'],
+            ['amundsen', 's.ocean']]
+
+
+@pytest.fixture
+def same_diff_keys_data():
+    """
+    Contains a mixture of same and distinct keys from 'std_test_data'
+    """
+    return [['norwegian', 'NORWAY'],
+            ['SCOTIA', 's.ocean'],
+            ['laptev', 'russia'],
+            ['Mawson', 'E.Antartica']]
+
+
+@pytest.fixture
+def combined_data():
+    """
+    The expected result of combining the 'std_test_data' &
+    'same_diff_keys_data' in a MetaDict
+    """
+    return [['labrador', 'americas'],
+            ['Norwegian', 'NORWAY'],
+            ['BALTIC', 'europe'],
+            ['LaPteV', 'russia'],
+            ['SCOTIA', 's.ocean'],
+            ['Mawson', 'E.Antartica']]
+
+
+@pytest.fixture
+def std_metadict(std_test_data):
+    return MetaDict(std_test_data)
+
+
+class TestInit:
```

Comment:
There is no need for these to be in classes when using pytest either. You should keep the docstrings as delimiters and comments, but pytest will collect all functions starting with `test_`.